liquibase-couchbase/src/main/java/liquibase/ext/couchbase/operator/ClusterOperator.java

```diff
@@ -80,6 +85,16 @@ public List<QueryIndex> getQueryIndexesForBucket(String bucketName) {
         return getQueryIndexes().getAllIndexes(bucketName);
     }
 
+    public List<String> retrieveDocumentIdsByWhereClause(Keyspace keyspace, String whereClause) {
+        String collectionPath = keyspace.getFullPath();
+        String documentIdRetrieveQuery = format(RETRIEVE_DOCUMENT_IDS_QUERY_TEMPLATE, collectionPath, whereClause);
+        QueryResult documentIdsResult = executeSingleSql(documentIdRetrieveQuery);
+        return documentIdsResult.rowsAsObject()
+                .stream()
+                .map(jsonObject -> jsonObject.getString("id"))
+                .collect(Collectors.toList());
+    }
+
```

Comment:
Maybe this should be in CollectionOperator?

Reply:
Surely I agree that the operation is related to collection, but to execute query we need cluster, and I think we can't execute query there without cluster. What you think?

Reply:
Exactly. I forgot about this :) 

So it's ok